engines/lilliput/detection.cpp

```diff
@@ -0,0 +1,286 @@
+				{"isomap.dta", 0, "bad97eae03a4db3e99565e39b0b3c06a", 16384},
+				AD_LISTEND
+			},
+			Common::EN_ANY,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		kGameTypeRobin
+	},
+	// Robin Hood French
+	{
+		{
+			"robin", 0,
+			{
+				{"frules.prg", 0, "cf076c5ebfe8b3571e74a6a46d79426f", 76660},
+				{"isomap.dta", 0, "bad97eae03a4db3e99565e39b0b3c06a", 16384},
+				AD_LISTEND
+			},
+			Common::FR_FRA,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		kGameTypeRobin
+	},
+	// Robin Hood German
+	{
+		{
+			"robin", 0,
+			{
+				{"grules.prg", 0, "b53b7353dc1e841b206a64851e7bc58c", 78050},
+				{"isomap.dta", 0, "bad97eae03a4db3e99565e39b0b3c06a", 16384},
+				AD_LISTEND
+			},
+			Common::DE_DEU,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		kGameTypeRobin
+	},
+	// Robin Hood Italian
+	{
+		{
+			"robin", 0,
+			{
+				{"irules.prg", 0, "4d69ed3cda1e1d73585905517ea705d1", 75654},
+				{"isomap.dta", 0, "bad97eae03a4db3e99565e39b0b3c06a", 16384},
+				AD_LISTEND
+			},
+			Common::IT_ITA,
+			Common::kPlatformDOS,
+			ADGF_UNSTABLE,
+			GUIO0()
+		},
+		kGameTypeRobin
+	},
+	{AD_TABLE_END_MARKER, kGameTypeNone}
+};
+
+class LilliputMetaEngine : public AdvancedMetaEngine {
+public:
+	LilliputMetaEngine() : AdvancedMetaEngine(gameDescriptions, sizeof(LilliputGameDescription), lilliputGames) {
+	}
+
+	const char *getName() const {
+		return "Lilliput";
+	}
+
+	const char *getOriginalCopyright() const {
+		return "Lilliput Engine copyright S.L.Grand, Brainware, 1991-1992";
+	}
+
+	bool createInstance(OSystem *syst, Engine **engine, const ADGameDescription *gd) const;
+	bool hasFeature(MetaEngineFeature f) const;
+
+	int getMaximumSaveSlot() const;
+	SaveStateList listSaves(const char *target) const;
+	SaveStateDescriptor querySaveMetaInfos(const char *target, int slot) const;
+	void removeSaveState(const char *target, int slot) const;
+};
+
+bool LilliputMetaEngine::createInstance(OSystem *syst, Engine **engine, const ADGameDescription *gd) const {
+	if (gd) {
+		*engine = new LilliputEngine(syst, (const LilliputGameDescription *)gd);
+		((LilliputEngine *)*engine)->initGame((const LilliputGameDescription *)gd);
+	}
+	return gd != 0;
+}
+
+bool LilliputMetaEngine::hasFeature(MetaEngineFeature f) const {
+	return
+		(f == kSupportsListSaves) ||
+		(f == kSupportsLoadingDuringStartup) ||
+		(f == kSupportsDeleteSave) ||
+		(f == kSavesSupportMetaInfo) ||
+		(f == kSavesSupportThumbnail) ||
+		(f == kSavesSupportCreationDate);
+}
+
+int LilliputMetaEngine::getMaximumSaveSlot() const {
+	return 99;
+}
+
+SaveStateList LilliputMetaEngine::listSaves(const char *target) const {
+	Common::SaveFileManager *saveFileMan = g_system->getSavefileManager();
+	Common::StringArray filenames;
+	Common::String pattern = target;
+	pattern += "-??.SAV";
+
+	filenames = saveFileMan->listSavefiles(pattern);
+	sort(filenames.begin(), filenames.end());   // Sort (hopefully ensuring we are sorted numerically..)
```

Comment:
`Common::sort(saveList.begin(), saveList.end(), SaveStateDescriptorSlotComparator());` can now be used so replace hope with certainty.